data-sources/wikipedia-wikidata/import_wikidata.sh

```diff
@@ -0,0 +1,88 @@
+#!/bin/bash
+
+psqlcmd() {
+     psql wikiprocessingdb
+}
+
+mysql2pgsqlcmd() {
+     ./mysql2pgsql.perl /dev/stdin /dev/stdout
+}
+
+
+# list the languages to process (refer to List of Wikipedias here: https://en.wikipedia.org/wiki/List_of_Wikipedias)
+
+language=( "ar" "bg" "ca" "cs" "da" "de" "en" "es" "eo" "eu" "fa" "fr" "ko" "hi" "hr" "id" "it" "he" "lt" "hu" "ms" "nl" "ja" "no" "pl" "pt" "kk" "ro" "ru" "sk" "sl" "sr" "fi" "sv" "tr" "uk" "vi" "vo" "war" "zh" )
+
+
+# get wikidata tables
+
+wget https://dumps.wikimedia.org/wikidatawiki/latest/wikidatawiki-latest-geo_tags.sql.gz
+wget https://dumps.wikimedia.org/wikidatawiki/latest/wikidatawiki-latest-page.sql.gz
+wget https://dumps.wikimedia.org/wikidatawiki/latest/wikidatawiki-latest-wb_items_per_site.sql.gz
+
+
+# import wikidata tables
+
+gzip -dc wikidatawiki-latest-geo_tags.sql.gz | mysql2pgsqlcmd | psqlcmd
+gzip -dc wikidatawiki-latest-page.sql.gz | mysql2pgsqlcmd | psqlcmd
+gzip -dc wikidatawiki-latest-wb_items_per_site.sql.gz | mysql2pgsqlcmd | psqlcmd
+
+
+# create derrived tables
+
+echo "CREATE TABLE geo_earth_primary AS SELECT gt_page_id, gt_lat, gt_lon FROM geo_tags WHERE gt_globe = 'earth' AND gt_primary = 1 AND NOT( gt_lat < -90 OR gt_lat > 90 OR gt_lon < -180 OR gt_lon > 180 OR gt_lat=0 OR gt_lon=0) ;" | psqlcmd
+echo "CREATE TABLE geo_earth_wikidata AS SELECT DISTINCT geo_earth_primary.gt_page_id, geo_earth_primary.gt_lat, geo_earth_primary.gt_lon, page.page_title, page.page_namespace FROM geo_earth_primary LEFT OUTER JOIN page ON (geo_earth_primary.gt_page_id = page.page_id) ORDER BY geo_earth_primary.gt_page_id;" | psqlcmd
+echo "CREATE TABLE geo_earth_wikidata_pages (gt_page_id integer, gt_lat numeric(11,8), gt_lon numeric(11,8), language text, page_title text, instance_of text, ips_site_page text);" | psqlcmd
+
+
+# get wikidata places from wikidata query API
+
+while read F  ; do
+    wget "https://query.wikidata.org/bigdata/namespace/wdq/sparql?format=json&query=SELECT ?item?lat?lon WHERE{?item wdt:P31*/wdt:P279*wd:$F;wdt:P625?pt.?item p:P625?loc.?loc psv:P625?cnode.?cnode wikibase:geoLatitude?lat.?cnode wikibase:geoLongitude?lon.}" -O $F.json
+    jq -r '.results | .[] | .[] | [.item.value,.lat.value, .lon.value] | @csv ' $F.json >> $F.txt
+    awk -v qid=$F '{print $0 ","qid}' $F.txt | sed -e 's!"http://www.wikidata.org/entity/!!' | sed 's/"//g' >> $F.csv
+    cat $F.csv >> wikidata_place_dump.csv
+    rm $F.json $F.txt $F.csv
+done < wikidata_place_types.txt
+
+
+# import wikidata places
+
+echo "CREATE TABLE wikidata_place_dump (item text,  lat numeric(11,8), lon numeric(11,8), instance_of text, ont_level integer);"  | psqlcmd
+echo "COPY wikidata_place_dump (item, lat, lon, instance_of) FROM  '/srv/nominatim/Nominatim/data-sources/wikipedia-wikidata/wikidata_place_dump.csv' DELIMITER ',' CSV;"  | psqlcmd
+echo "CREATE TABLE wikidata_places AS SELECT DISTINCT item, instance_of, lon, lat, ont_level FROM wikidata_place_types;" | psqlcmd
```

Comment:
What creates the `wikidata_place_types` table?

Reply:
We need instructions how this table was created. On a fresh system the script fails because it expects the table. On the gsoc2019 server the table seems to have 11m entries.

Reply:
thanks a lot, works now